main/src/library/Console.flix

```diff
@@ -50,6 +50,24 @@ mod Console {
 
     import java.lang.System
 
+    ///
+    /// Reads a 32-bit integer from the console.
+    ///
+    pub def readInt32(): Option[Int32] \ Console =
+        Console.readln() |> Int32.fromString
+
+    ///
+    /// Reads a 32-bit integer from the console with a prompt.
+    ///
+    /// Repeatedly prints `msg` and reads input until a valid integer in the given range `[start; end)` is entered.
+    ///
+    pub def promptInt32(start: {start = Int32}, end: {end = Int32}, msg: a): Int32 \ Console with ToString[a] =
```

Comment:
I think this pattern encourages bad UX.

It's rare that the right thing to do is prompt the user with the same exact message without telling them what's wrong with the original.

Reply:
I don't love the msg being `a` either. Seems fishy to prompt the user with e.g. a Boolean.

Reply:
> I think this pattern encourages bad UX.

Also my thinking, the current code also has no way to quit the loop or cancel.

In the generalized version `parse: String -> Result[String, t]` could be used to give feedback